### ISSUE TITLE
Non-functioning constraints in .getrightneighbours(), .getcooc() etc.

### ISSUE THREAD

**Issue by KasperFyhn** [proycon/colibri-core#44]
I wanted to get only n-grams of a specific size following some other n-gram. However, I experienced that the output did not adhere to the given constraints, at least not as I expected. I've consulted the documentation to figure out if I simply misunderstood something; if so, please enlighten me. :-)

Here is a working example (most of it taken from the tutorial notebook) which shows it:

```python
import colibricore
from urllib.request import urlopen


TMPDIR = '/tmp/'
corpusfile_plato_plaintext = TMPDIR + "republic.txt"
classfile_plato = TMPDIR + "republic.colibri.cls"
corpusfile_plato = TMPDIR + "republic.colibri.dat"

f = urlopen('http://lst.science.ru.nl/~proycon/republic.txt')
with open(corpusfile_plato_plaintext,'wb') as of:
    of.write(f.read())
print("Downloaded to " + corpusfile_plato_plaintext)

# make encoder, encode corpus and make decoder
classencoder = colibricore.ClassEncoder(classfile_plato)
classencoder.build(corpusfile_plato_plaintext)
classencoder.save(classfile_plato)
classencoder.encodefile(corpusfile_plato_plaintext, corpusfile_plato)
classdecoder = colibricore.ClassDecoder(classfile_plato)

# set options and train model
options = colibricore.PatternModelOptions(mintokens=2, maxlength=8,
                                          doskipgrams=True)
corpus_plato = colibricore.IndexedCorpus(corpusfile_plato)
model = colibricore.IndexedPatternModel(reverseindex=corpus_plato)
model.train(corpusfile_plato, options)

# make ngram and get its neighbours under different constraints
ngram = classencoder.buildpattern("the law")
no_constraint = {(pattern, count)
                 for pattern, count in model.getrightneighbours(ngram, 1)}

only_bigrams = {(pattern, count)
                 for pattern, count in model.getrightneighbours(ngram, 1, size=2)}

# we'd expect nothing besides bigrams, but ...
for pattern, count in only_bigrams:
    if not pattern.isskipgram() and len(pattern) != 2:
        print('Found a non-bigram where I should not!: ',
              pattern.tostring(classdecoder))
        break

only_ngrams = {(pattern, count)
               for pattern, count in model.getrightneighbours(
        ngram, 1, category=colibricore.Category.NGRAM
    )}
# we'd expect no skipgrams, but ...
for pattern, count in only_ngrams:
    if pattern.isskipgram():
        print('Found a skipgram where I should not!',
              pattern.tostring(classdecoder))
        break

```
Output:
```
Found a non-bigram where I should not!:  ; at the same time
Found a skipgram where I should not! ; {*} their
```

Similar things happen for cooc methods and left neighbours.

**Comment by proycon** [comment]
Thanks for the bug report with code to reproduce the problem! You are right, that output looks wrong indeed. There seems to be something going wrong internally. I'll have to investigate what caused it as I haven't seen this before.

**Comment by proycon** [comment]
I found the cause of the problem and have corrected it in the newly released Colibri Core 2.5.2. This should solve your problem!
